Description: Close the right page when a child process exits
 When a child process exits, sakura closes the current notebook page,
 which can be different from the page in which the child process was
 actually running.
Author: Thibaut Girka <thib@sitedethib.com>
Forwarded: yes
Bug: https://bugs.launchpad.net/sakura/+bug/984493 
Debian-Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=668988

---
 src/sakura.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: sakura.git/src/sakura.c
===================================================================
--- sakura.git.orig/src/sakura.c	2012-05-01 09:48:03.029589392 +0900
+++ sakura.git/src/sakura.c	2012-05-01 09:48:19.581580620 +0900
@@ -1528,7 +1528,8 @@
 	struct terminal *term;
 	gint page, npages;
 
-	page = gtk_notebook_get_current_page(GTK_NOTEBOOK(sakura.notebook));
+	page = gtk_notebook_page_num(GTK_NOTEBOOK(sakura.notebook),
+				gtk_widget_get_parent(widget));
 	npages = gtk_notebook_get_n_pages(GTK_NOTEBOOK(sakura.notebook));
 	term = sakura_get_page_term(sakura, page);
 
